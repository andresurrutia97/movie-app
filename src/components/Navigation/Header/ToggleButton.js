import React from 'react';
import PropTypes from 'prop-types';

const ToggleButton = ({ clicked }) => {
  return (
    <div className="block absolute left-0 md:hidden">
      <button
        onClick={clicked}
        type="button"
        className="flex items-center text-white"
      >
        <span className="material-icons">menu</span>
      </button>
    </div>
  );
};

ToggleButton.propTypes = {
  clicked: PropTypes.func.isRequired,
};

export default ToggleButton;
