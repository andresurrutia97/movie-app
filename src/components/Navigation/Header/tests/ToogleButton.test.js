import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ToggleButton from '../ToggleButton';

describe('<ToggleButton />', () => {
  const mockFn = jest.fn();

  it('should render correctly', () => {
    const tree = renderer.create(<ToggleButton clicked={mockFn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call the mockFn function when click', () => {
    const wrapper = shallow(<ToggleButton clicked={mockFn} />);
    wrapper.find('button').simulate('click');
    expect(mockFn).toBeCalledTimes(1);
  });
});
