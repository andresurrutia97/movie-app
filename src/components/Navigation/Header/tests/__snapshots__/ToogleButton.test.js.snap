// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> should render correctly 1`] = `
<div
  className="block absolute left-0 md:hidden"
>
  <button
    className="flex items-center text-white"
    onClick={[MockFunction]}
    type="button"
  >
    <span
      className="material-icons"
    >
      menu
    </span>
  </button>
</div>
`;
